server/server.go

```diff
@@ -114,8 +120,20 @@ var backoff = wait.Backoff{
 var (
 	clientConstraint = fmt.Sprintf(">= %s", common.MinClientVersion)
 	baseHRefRegex    = regexp.MustCompile(`<base href="(.*)">`)
+	// limits number of concurrent login requests to prevent password brute forcing. If set to 0 then no limit is enforced.
+	maxConcurrentLoginRequestsCount = 50
 )
 
+func init() {
+	if valStr := os.Getenv(maxConcurrentLoginRequestsCountEnv); valStr != "" {
```

Comment:
Hm, maybe we should move `parseNumFromEnv()` from `util/session` (https://github.com/argoproj/argo-cd/blob/7660e40fdcd1d870528f118a0618a69a8331426a/util/session/sessionmanager.go#L114) to a generic util function so we can reuse it whenever we need to parse a number from an environment variable.

Reply:
good suggestion! done.